File: app/src/main/java/by/home/dartlen/dindindon/pendingalarms/util/AlarmAdapter.java
package by.home.dartlen.dindindon.pendingalarms.util;

import android.text.format.DateFormat;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;


import java.util.List;

import androidx.recyclerview.widget.RecyclerView;
import by.home.dartlen.dindindon.R;


public class AlarmAdapter extends RecyclerView.Adapter<AlarmAdapter.ViewHolder> implements View.OnClickListener {

    private List<Alarm> mAlarmList;
    private final NotesAdapterInteraction mListener;

    public AlarmAdapter(NotesAdapterInteraction notesAdapterInteraction, List<Alarm> alarmList) {
        mListener = notesAdapterInteraction;
        mAlarmList = alarmList;
    }


    public static class ViewHolder extends RecyclerView.ViewHolder {
        public final View mView;
        public final TextView tvTime;


        public ViewHolder(View view) {
            super(view);
            mView = view;

            tvTime = view.findViewById(R.id.tv_alarm);

        }
    }

    @Override
    public AlarmAdapter.ViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.item_note, parent, false);
        return new ViewHolder(view);
    }

    @Override
    public void onBindViewHolder(AlarmAdapter.ViewHolder holder, int position) {
        holder.tvTime.setText(DateFormat.format("dd-MM-yyyy/HH:mm",(mAlarmList.get(position).getTime()*1000)).toString());
         holder.mView.setOnClickListener(this);
        holder.mView.setTag(position);
    }

    @Override
    public void onClick(View view) {
        int position = (int) view.getTag();
        Alarm alarm = mAlarmList.get(position);
        mListener.onClickItem(alarm);
        //Опции
       // notifyItemRemoved(position);
       // notifyItemRangeChanged(position, mAlarmList.size());
    }

    public void updateList(List<Alarm> updatedList) {
        mAlarmList = updatedList;
        notifyDataSetChanged();
    }


    @Override
    public int getItemCount() {
        return mAlarmList.size();
    }

    public interface NotesAdapterInteraction {
        void onClickItem(Alarm alarm);
    }
}

File: app/src/main/java/by/home/dartlen/dindindon/pendingalarms/util/AlarmAdapter.kt
package by.home.dartlen.dindindon.pendingalarms.util

import android.annotation.SuppressLint
import android.text.format.DateFormat
import android.view.View
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import by.home.dartlen.dindindon.R
import by.home.dartlen.dindindon.inflate
import kotlinx.android.synthetic.main.item_alarm.view.*


class AlarmAdapter(private val listener: (Alarm) -> Unit) : RecyclerView.Adapter<RecyclerView.ViewHolder>() {
    var list: ArrayList<Alarm>? = arrayListOf()
    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): RecyclerView.ViewHolder {
        return ViewHolder(parent.inflate(R.layout.item_alarm))
    }

    override fun onBindViewHolder(holder: RecyclerView.ViewHolder, position: Int) {
        (holder as ViewHolder).bind(list!![position], listener)
    }

    override fun getItemCount(): Int {
        return list!!.size
    }

    inner class ViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        @SuppressLint("SetTextI18n")
        fun bind(item: Alarm, listener: (Alarm) -> Unit) {
            /*itemView.geo.text = item.lng.toString() + "," + item.lon.toString()
            itemView.city.text = item.city
            itemView.date.text = DateUtils.formatDateTime(itemView.context, DateTime(item.createdAt), DateUtils.FORMAT_SHOW_DATE or DateUtils.FORMAT_SHOW_TIME)*/
            itemView.tv_alarm.text = DateFormat.format("dd-MM-yyyy/HH:mm", (item.time * 1000)).toString()
            itemView.setOnClickListener {
                listener(item)
            }
        }
    }
}